### PR TITLE
Add optional requirements to a txt file for Docker caching

### LINKED ISSUE #478: Add the optional dependencies to a requirements file for caching in Docker
### Is your feature request related to a specific problem?

Right now the command for `make cli` takes longer and longer because of the optional dependencies getting reinstalled every time there is a code change and the docker file gets built.

### Describe the solution you'd like

Have a file where we list the optional requirements so that we can install them at an earlier step, before `pip install -e .`

### Describe alternatives you've considered, if any

Leaving it as-is

### Additional context

This would require having duplicated requirements, but these don't change often and the speed-up is worth the small trade off to me

### PR DESCRIPTION
**Pull request by ThomasLaPiana**
Closes #478 

### Code Changes

* [x] add a file for optional dependencies
* [x] Update the dockerfile to copy and install these before `pip install -e .`
* [x] update the PR checks to have discrete steps for the unit/integration/external tests

### Steps to Confirm

* [x] all of the checks pass
* [x] local commands work as normal
* [x] local builds take much less time

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created

### Description Of Changes

Two small things are changing in this PR for general developer experience improvement:
1. The optional dependencies will now get installed _before_ the `pip install -e` step in the Dockerfile, this should greatly speed up any make commands that rely on the Dockerfile
1. Split the different pytest invocations out into their own steps. This should be faster (integrations don't have to wait for unit, etc.) and also make it more transparent which tests are breaking

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
new build process took 13.1 seconds when the requirements were cached via `optional_requirements.txt`
old build process took 35.1 seconds when the requirements were loaded at `pip install -e .` command

This gap will grow much wider when dealing with less resources, definitely worth the change

File: .github/workflows/pr_checks.yml [hunk 1]
```diff
@@ -160,9 +160,41 @@ jobs:
       - name: Run unit test suite
         run: make pytest-unit
 
+  Pytest-Integration:
```
Comment (SteveDMurphy) [inline .github/workflows/pr_checks.yml:163]: Excellent idea 🚄

File: fidesctl/setup.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 dev_requires = open("dev-requirements.txt").read().strip().split("\n")
 
 # Human-Readable/Reusable Extras
+# Add these to `optional_requirements.txt` as well
```
Comment (SteveDMurphy) [inline fidesctl/setup.py:15]: It took me a second to realize this is a reminder for future dependencies we add, thank you for adding 💙

**Review by SteveDMurphy — APPROVED** [review]
LGTM! Confirmed the docker build seems to be ⚡ ⚡ ⚡ in comparison 💥

**Review by PSalant726 — APPROVED** [review]
This is a great improvement!

**Pull request merged**